keep/providers/auth0_provider/auth0_provider.py

```diff
@@ -0,0 +1,147 @@
+"""
+Auth0 provider.
+"""
+import dataclasses
+import os
+from typing import Optional
+
+import requests
+from pydantic.dataclasses import dataclass
+
+from keep.providers.base.base_provider import BaseProvider
+from keep.providers.models.provider_config import ProviderConfig
+
+
+@dataclasses.dataclass
+class Auth0ProviderAuthConfig:
+    """
+    Auth0 authentication configuration.
+    """
+
+    token: str = dataclasses.field(
+        default=None,
+        metadata={
+            "required": True,
+            "description": "Auth0 api token",
+            "hint": "https://manage.auth0.com/dashboard/us/YOUR TENANT/apis/management/explorer",
+        },
+    )
+    domain: str = dataclasses.field(
+        default=None,
+        metadata={
+            "required": True,
+            "description": "Auth0 domain",
+        },
+    )
+
+
+class Auth0Provider(BaseProvider):
+    provider_id: str
+    config: ProviderConfig
+
+    def __init__(self, provider_id: str, config: ProviderConfig):
+        super().__init__(provider_id, config)
+
+    def validate_config(self):
+        """
+        Validates required configuration for Auth0 provider.
+
+        """
+        if self.config.authentication is None:
+            self.config.authentication = {}
+        self.authentication_config = Auth0ProviderAuthConfig(
+            **self.config.authentication
+        )
+
+    def notify(self, **kwargs):
+        pass  # Define how to notify about any alerts or issues
+
+    def dispose(self):
+        pass
+
+
+class Auth0LogsProvider(Auth0Provider):
```

Comment:
Doesn't really seem to let you query logs but rather more specific to new users?

Reply:
It does (you query by log type)